### ISSUE TITLE
Potential stack overflow on Q1

### ISSUE THREAD

**Issue by gpollatos** [takama/task#1]
Given the somewhat limited supported in current Go versions for tail call optimization this function can produce a stack overflow given sufficiently large input.

**Comment by takama** [comment]
Yes, you are right, this is not practical solution, just for fun, as well I lost `*` operation :) We always can translate it in loop, it would look bored :)
